dev-docs/how-to-contribute.adoc

```diff
@@ -0,0 +1,23 @@
+# How To Contribute
+
+The Apache Solr Github Repository is located here: https://github.com/apache/solr. Typically all contributions to Solr are developed on a fork of the Solr repository, then opened as PRs against the `main` branch in the solr repository. These PRs are then linked to an open JIRA issue and members of the community will review, comment, approve, and commit changes.
+
+Please keep in mind that the Solr is community driven with many members taking their own personal time to help respond to issue and review PRs. Be open to feedback and discussions and maintain a respectful approach.
+
+## Setup for Contributing
+In order to get setup you will need a fork of the Apache Solr repository and an account in the ASF Jira (usable for all Apache projects).
+
+1. Create a fork of the https://github.com/apache/solr[Solr repository] into your own personal account or a your corporate organization account using the Github instructions provided here: https://docs.github.com/en/get-started/quickstart/fork-a-repo.
+2. Create an account in the ASF Jira: https://issues.apache.org/jira/login.jsp which you will use to create new issues.
+
+## Making a New Contribution
+In order to make a new contribution to Solr you will use the fork you have created above.
+
+1. Create a new Jira issue in the Solr project: https://issues.apache.org/jira/projects/SOLR/issues
+2. Create a new branch in your Solr fork to provide a PR for your contribution on the newly created issue. Make any necessary changes for the given bug/feature in that branch. You can use additional information in these dev-docs to build and test your code as well as ensure it passes code quality checks.
+3. Once you are satisfied with your changes, open a PR of your branch against the `main` branch of the apache/solr repository. When you open a PR on your fork, this should be the default option.
+* The title of your PR should include the Solr Jira issue that you opened, i.e. `SOLR-12345: New feature`.
+* The PR description will automatically populate with a pre-set template that you will need to fill out.
+* The PR will run a gradle pre-commit task that will do some basic checks of your changes. This is run as a Github Action and the results can be viewed from your PR.
+* If you properly named your PR you will see it automatically linked to your Jira issue when you view the issue.
+4. At this point, you will need a Solr committer to review your change and help to get it merged upstream. You can use the https://solr.apache.org/community.html[Solr Developers Mailing List] to get attention to your issue.
```

Comment:
Should we mention PR draft state? It can be useful if you know there is more work to do...